library/Director/Objects/IcingaServiceSet.php

```diff
@@ -327,4 +327,29 @@ protected function beforeStore()
             );
         }
     }
+
+    public function onSingleIcingaConfig(IcingaConfig $config)
```

Comment:
I do not like the name of this method. As you're calling parent:: anyways, couldn't you just override `toSingleIcingaConfig()` to get the same result? It would duplicate the try/catch part, but a) who cares or b) move that part to a `renderFailsafe($callback)` method.